src/components/CategoryFilter.js

```diff
@@ -0,0 +1,30 @@
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { CATEGORIES } from '../containers/BooksForm';
+import { changeFilter } from '../redux/actions';
+
+const allCategories = [{ name: 'All', id: '0' }, ...CATEGORIES];
+
+const CategoryFilter = () => {
+  const [category, setCategory] = useState('0');
+  const dispatch = useDispatch();
+
+  const handleChange = (event) => {
+    const categoryName = allCategories.find((cat) => cat.id === event.target.value).name;
+    setCategory(event.target.value);
+    dispatch(changeFilter(categoryName));
+  };
```

Comment:
- [ ] The redux store already stores the category therefore please remove this react state there is no need for it.